lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -3907,6 +3927,7 @@ private static void carryOverHardDeletes(ReadersAndUpdates mergedReadersAndUpdat
   @SuppressWarnings("try")
   private synchronized boolean commitMerge(MergePolicy.OneMerge merge, MergeState mergeState) throws IOException {
```

Comment:
We should really rename this method to not use the word `commit` since that already means something (different!) in `IndexWriter`.  Maybe `finishMerge`, or `applyMergedSegments` or something.  Naming is the hardest part!  We don't have to do that in this PR...